From 0ddefa8bdbf921247b14f9e8d30ea26f0ae3337b Mon Sep 17 00:00:00 2001
From: rnaidu <ramya.v.naidu@intel.com>
Date: Tue, 28 May 2019 09:53:11 +0530
Subject: [PATCH] To load Swiftshader, we need to make changes in
 frameworks/native Loader.cpp to forcefully pick the GFx module

Signed-off-by: rnaidu <ramya.v.naidu@intel.com>
Signed-off-by: sgnanase <sundar.gnanasekaran@intel.com>
---
 opengl/libs/EGL/Loader.cpp | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/opengl/libs/EGL/Loader.cpp b/opengl/libs/EGL/Loader.cpp
index 91a34558a..81b577919 100644
--- a/opengl/libs/EGL/Loader.cpp
+++ b/opengl/libs/EGL/Loader.cpp
@@ -225,7 +225,7 @@ void* Loader::open(egl_connection_t* cnx)
 
     setEmulatorGlesValue();
 
-    dso = load_driver("GLES", cnx, EGL | GLESv1_CM | GLESv2);
+    dso = load_driver("EGL", cnx, EGL | GLESv1_CM | GLESv2);
     if (dso) {
         hnd = new driver_t(dso);
     } else {
@@ -326,20 +326,7 @@ static void* load_system_driver(const char* kind) {
             int emulationStatus = checkGlesEmulationStatus();
             switch (emulationStatus) {
                 case 0:
-#if defined(__LP64__)
-                    result = "/vendor/lib64/egl/libGLES_android.so";
-#else
-                    result = "/vendor/lib/egl/libGLES_android.so";
-#endif
-                    return result;
                 case 1:
-                    // Use host-side OpenGL through the "emulation" library
-#if defined(__LP64__)
-                    result = std::string("/vendor/lib64/egl/lib") + kind + "_emulation.so";
-#else
-                    result = std::string("/vendor/lib/egl/lib") + kind + "_emulation.so";
-#endif
-                    return result;
                 case 2:
                     // Use guest side swiftshader library
 #if defined(__LP64__)
-- 
2.21.0

